src/Promitor.Core.Scraping/Configuration/Serialization/Deserializer.cs

```diff
@@ -37,5 +38,81 @@ internal List<TObject> Deserialize(YamlSequenceNode nodes)
 
             return deserializedObjects;
         }
+
+        /// <summary>
+        /// Gets the string value of the specified yaml property.
+        /// </summary>
+        /// <param name="node">The node containing the property.</param>
+        /// <param name="propertyName">The name of the property.</param>
+        /// <returns>The string value of the property.</returns>
+        protected static string GetString(YamlMappingNode node, string propertyName)
+        {
+            if (node.Children.TryGetValue(propertyName, out var propertyNode))
+            {
+                return propertyNode.ToString();
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the value of the specified yaml property converted to an enum.
+        /// </summary>
+        /// <typeparam name="T">The type of enum to return.</typeparam>
+        /// <param name="node">The node containing the property.</param>
+        /// <param name="propertyName">The property name.</param>
+        /// <returns>The enum value, or null if the property doesn't exist.</returns>
+        protected static T? GetEnum<T>(YamlMappingNode node, string propertyName)
+            where T : struct
+        {
+            if (node.Children.TryGetValue(propertyName, out var propertyNode))
+            {
+                if (System.Enum.TryParse<T>(propertyNode.ToString(), out var enumResult))
+                {
+                    return enumResult;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the contents of the specified property as a dictionary.
+        /// </summary>
+        /// <param name="node">The node containing the property.</param>
+        /// <param name="propertyName">The name of the property.</param>
+        /// <returns>The child items of the property as a dictionary.</returns>
+        protected static Dictionary<string, string> GetDictionary(YamlMappingNode node, string propertyName)
+        {
+            if (node.Children.TryGetValue(propertyName, out var propertyNode))
+            {
+                var result = new Dictionary<string, string>();
+
+                foreach (var (key, value) in ((YamlMappingNode)propertyNode).Children)
+                {
+                    result[key.ToString()] = value.ToString();
+                }
+
+                return result;
+            }
+
+            return null;
```

Comment:
Personally I'd avoid prematurely optimising - at the moment there's no flag because we don't use the functionality. If we add code that needs to alter the behaviour we can add a flag at that point, otherwise we've got code that's never used.

Does that make sense?

Reply:
Sure that makes perfect sense, was just wondering if we need it already or not. If that is not the case, feel free to just resolve!